### PR TITLE
fixed build dedup bug caused by recent change.

### PR DESCRIPTION
**Pull request by heejaechang**

### CODE REVIEW

File: src/VisualStudio/Core/Def/Implementation/TaskList/ExternalErrorDiagnosticUpdateSource.cs [hunk 1]
```diff
@@ -437,11 +441,13 @@ public IEnumerable<ProjectId> GetProjectsWithoutErrors(Solution solution)
             public ImmutableDictionary<ProjectId, ImmutableArray<DiagnosticData>> GetLiveDiagnosticsPerProject(Func<DiagnosticData, bool> liveDiagnosticChecker)
             {
                 var builder = ImmutableDictionary.CreateBuilder<ProjectId, ImmutableArray<DiagnosticData>>();
-                foreach (var projectKv in _projectMap)
+                foreach (var projectId in GetProjectIds())
                 {
-                    // get errors that can be reported by live diagnostic analyzer
-                    var diagnostics = ImmutableArray.CreateRange(projectKv.Value.Concat(_documentMap.Where(kv => kv.Key.ProjectId == projectKv.Key).SelectMany(kv => kv.Value)).Where(liveDiagnosticChecker));
-                    builder.Add(projectKv.Key, diagnostics);
+                    var diagnostics = ImmutableArray.CreateRange(
```
Comment (heejaechang) [inline src/VisualStudio/Core/Def/Implementation/TaskList/ExternalErrorDiagnosticUpdateSource.cs]: this is actual change, others are just clean up.

**Comment by heejaechang** [comment]
@mavasani can you take a look?

**Comment by srivatsn** [comment]
:+1:

**Comment by mavasani** [comment]
👍

**Comment by heejaechang** [comment]
retest vsi please

**Comment by mavasani** [comment]
@heejaechang once this is merged in can you uncomment the skipped scenario in VSI test CSharpErrorLog.xml?

**Comment by heejaechang** [comment]
@mavasani  that require - https://github.com/dotnet/roslyn-internal/pull/829 - as well.

**Comment by mavasani** [comment]
I am not sure - CSharpErrorLog.xml uses a checked in test solution and doesn't attempt to make any changes to the project file.

**Comment by heejaechang** [comment]
@mavasani ah, sorry confused with CSharpErrorList.xml :) for CSharpErrorLog.xml I need https://github.com/dotnet/roslyn/issues/10346 to be fixed. currently working on.

**Comment by mavasani** [comment]
So was I :) I thought this was fix for #10346

**Pull request merged**